dropwizard-health/src/test/java/io/dropwizard/health/response/DetailedJsonHealthResponseProviderTest.java

```diff
@@ -0,0 +1,253 @@
+            new HealthStateView("baz", false, HealthCheckType.READY, false));
+        final Collection<String> names = views.stream().map(HealthStateView::getName).collect(Collectors.toList());
+
+        // when
+        views.forEach(view -> when(healthStateAggregator.healthStateView(view.getName()))
+            .thenReturn(Optional.of(view)));
+        when(healthStatusChecker.isHealthy(isNull())).thenReturn(true);
+        final HealthResponse response = detailedJsonHealthResponseProvider.partialHealthResponse(null, names);
+
+        // then
+        assertThat(response.isHealthy()).isTrue();
+        assertThat(response.getMessage()).hasValueSatisfying(message ->
+            assertThat(message).isEqualToIgnoringWhitespace(fixture("json/multiple-healthy-responses.json"))
+        );
+        assertThat(response.getContentType()).isEqualTo(MediaType.APPLICATION_JSON);
+    }
+
+    @Test
+    void partialHealthResponseShouldHandleZeroHealthStateViewCorrectly() {
+        // given
+        // when
+        final String name = "foo";
+        when(healthStatusChecker.isHealthy(isNull())).thenReturn(false);
+        when(healthStateAggregator.healthStateView(name)).thenReturn(Optional.empty());
+        final HealthResponse response = detailedJsonHealthResponseProvider.partialHealthResponse(null,
+            Collections.singleton(name));
+
+        // then
+        assertThat(response.isHealthy()).isFalse();
+        assertThat(response.getMessage()).hasValueSatisfying(message ->
+            assertThat(message).isEqualToIgnoringWhitespace("[]")
+        );
+        assertThat(response.getContentType()).isEqualTo(MediaType.APPLICATION_JSON);
+    }
+
+    @Test
+    void partialHealthResponseShouldThrowExceptionWhenJsonProcessorExceptionOccurs() throws IOException {
+        // given
+        final ObjectMapper mapperMock = mock(ObjectMapper.class);
+        this.detailedJsonHealthResponseProvider = new DetailedJsonHealthResponseProvider(healthStatusChecker,
+            healthStateAggregator, mapperMock);
+        final HealthStateView view =
+            new HealthStateView("foo", true, HealthCheckType.READY, true);
+        final JsonMappingException exception = JsonMappingException.fromUnexpectedIOE(new IOException("uh oh"));
+
+        // when
+        when(healthStateAggregator.healthStateView(eq(view.getName()))).thenReturn(Optional.of(view));
+        when(mapperMock.writeValueAsString(any()))
+            .thenThrow(exception);
+
+        // then
+        assertThatThrownBy(() -> detailedJsonHealthResponseProvider.partialHealthResponse(null,
+            Collections.singleton(view.getName())))
+            .isInstanceOf(RuntimeException.class)
+            .hasCauseReference(exception);
+        verifyNoInteractions(healthStatusChecker);
+    }
+
+    // Duplicated from dropwizard-testing due to circular deps
+    private String fixture(final String filename) {
```

Comment:
we can't depend on `dropwizard-testing` even after most of the components got pushed out of `dropwizard-core`?

Reply:
Nope, because there's still that circular dependency.

Core pulls in health, and testing pulls in core.

Reply:
gotcha.